datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -2224,6 +2354,179 @@ mod tests {
         assert_batches_sorted_eq!(expected, &batches);
     }
 
+    struct Coordinator {
```

Comment:
Usage example here. In a real-world scenario the `Coordinator` would likely be an external service which we communicate with through an rpc call